src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -456,6 +460,85 @@ class spell_sha_earthen_rage_proc_aura : public AuraScript
     }
 };
 
+// 61882 - Earthquake
+class spell_sha_earthquake : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_AFTERSHOCK, SPELL_SHAMAN_AFTERSHOCK_ENERGIZE });
+    }
+
+    void HandleAfterCast()
+    {
+        int32 energize = *GetSpell()->GetPowerTypeCostAmount(POWER_MAELSTROM);
```

Comment:
This is supposed to already be handled by `spell_sha_aftershock`, if it isn't working then it must be investigated why the proc doesn't work

Reply:
Aftershock cannot proc with triggered spell. Should I correct this in spell_proc table? Cause when Earthquake is cast as non-triggered there are 5 ticks instead of 6.

Reply:
But this is the spell that player directly casts from spellbook, that is never triggered

Reply:
Sorry, I've confused it with damage [Earthquake](https://www.wowhead.com/spell=77478/earthquake).
Aftershock doesn't proc due to mismatched proc type. It's `PROC_FLAG_DONE_SPELL_MAGIC_DMG_CLASS_POS` for earthquake and `PROC_FLAG_DONE_SPELL_MAGIC_DMG_CLASS_NEG` for aftershock. 
Will see what can be changed to make it work.